packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js

```diff
@@ -218,37 +221,91 @@ const InfoIndicatorButton = ({
-      },
     }
 
-    const buildStatusAction = buildStatusActions[buildStatus]
+    // this is because the build status enum is used for ui state - so we can't have 1 ui state that covers multiple build statuses.
+    // If we don't have content sync info (pollForNodeManifest), we have to assume that a currently building build isn't applicable to the current user. So we default to the least noisy state which is UPTODATE.
+    // @todo refactor and decouple build states from UI states - this state should be something like uiState === `IDLE`
+    buildStatusActions[BuildStatus.BUILDING] =
+      buildStatusActions[BuildStatus.UPTODATE]
+
+    const contentSyncSuccessAction = contentSyncRedirectUrl
+      ? () => {
+          setButtonProps(btnProps => {
+            return {
+              ...btnProps,
+              tooltip: {
+                testId: btnProps.testId,
+                content: (
+                  <BuildSuccessTooltipContent
+                    contentSyncRedirectUrl={contentSyncRedirectUrl}
+                  />
+                ),
+                closable: true,
+                onClose: closeInfoTooltip,
+              },
+              active: true,
+              hoverable: true,
+              showSpinner: false,
+            }
+          })
+        }
+      : false
+
+    const contentSyncLoadingAction = usingContentSync
+      ? () => {
+          setButtonProps(btnProps => {
+            return {
+              ...btnProps,
+              tooltip: {
+                testId: btnProps.testId,
+                content: `Building a new preview`,
+                overrideShow: true,
+              },
+              active: true,
+              hoverable: true,
+              showSpinner: true,
+            }
+          })
+        }
+      : false
+
+    const contentSyncErrorAction = contentSyncErrorMessage
+      ? buildStatusActions[BuildStatus.ERROR]
+      : false
+
+    const contentSyncAction =
+      contentSyncSuccessAction ||
+      contentSyncErrorAction ||
+      contentSyncLoadingAction
+
+    const buildStatusAction =
+      // with the introduction of Content Sync eager redirects, button state is no longer tied to build status. So we need a separate action for Content Sync.
+      contentSyncAction || buildStatusActions[buildStatus]
+
     if (typeof buildStatusAction === `function`) {
       buildStatusAction()
     } else {
       setButtonProps({ ...buttonProps, active: true, hoverable: true })
     }
-  }, [buildStatus, shouldShowFeedback])
+  }, [
+    buildStatus,
+    shouldShowFeedback,
+    contentSyncRedirectUrl,
+    contentSyncErrorMessage,
+  ])
 
   return (
     <IndicatorButton
       {...buttonProps}
       onClick={onInfoClick}
       onMouseEnter={buttonProps?.active ? trackHover : undefined}
       onTooltipToogle={updateTootipVisibility}
-      iconSvg={showNotificationInfoIcon ? infoAlertIcon : infoIcon}
+      iconSvg={
+        showNotificationInfoIcon || contentSyncRedirectUrl
```

Comment:
we need more creative check here. this should evaluate as false when we are in `contentSyncLoadingAction`.

Reply:
Thinking about this again once we have a redirect url we're no longer loading, so this might be ok as-is. If we have a redirect url, we prefer that over other states